runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -1417,3 +1417,113 @@ TR_J9ServerVM::dereferenceStaticFinalAddress(void *staticAddress, TR::DataType a
    return it->second;
    }
 
+bool
+TR_J9SharedCacheServerVM::isClassLibraryMethod(TR_OpaqueMethodBlock *method, bool vettedForAOT)
+   {
+   TR_ASSERT(vettedForAOT, "The TR_J9SharedCacheServerVM version of this method is expected to be called only from isClassLibraryMethod.\n"
+                                          "Please consider whether you call is vetted for AOT!");
+
+   if (getSupportsRecognizedMethods())
+      return this->TR_J9ServerVM::isClassLibraryMethod(method, vettedForAOT);
+
+   return false;
+   }
+
+bool
+TR_J9SharedCacheServerVM::stackWalkerMaySkipFrames(TR_OpaqueMethodBlock *method, TR_OpaqueClassBlock *methodClass)
+   {
+   bool skipFrames = TR_J9ServerVM::stackWalkerMaySkipFrames(method, methodClass);
+   TR::Compilation *comp = TR::comp();
+   if (comp && comp->getOption(TR_UseSymbolValidationManager))
+      {
+      if (!comp->getSymbolValidationManager()->addStackWalkerMaySkipFramesRecord(method, methodClass, skipFrames))
+         {
+         TR_ASSERT(false, "Failed to validate addStackWalkerMaySkipFramesRecord\n");
+         }
+      }
+   return skipFrames;
+   }
+
+bool
+TR_J9SharedCacheServerVM::isMethodEnterTracingEnabled(TR_OpaqueMethodBlock *method)
+   {
+   // We want to return the same answer as TR_J9VMBase unless we want to force it to allow tracing
+   return TR_J9VMBase::isMethodEnterTracingEnabled(method) || TR::Options::getAOTCmdLineOptions()->getOption(TR_EnableAOTMethodEnter);
+   }
+
+bool
+TR_J9SharedCacheServerVM::isMethodExitTracingEnabled(TR_OpaqueMethodBlock *method)
+   {
+   return TR_J9VMBase::isMethodExitTracingEnabled(method) || TR::Options::getAOTCmdLineOptions()->getOption(TR_EnableAOTMethodExit);
+   }
+
+bool
+TR_J9SharedCacheServerVM::traceableMethodsCanBeInlined()
+   {
+   return true;
+   }
+
+bool
+TR_J9SharedCacheServerVM::canMethodEnterEventBeHooked()
+   {
+   // We want to return the same answer as TR_J9ServerVM unless we want to force it to allow tracing
+   return TR_J9ServerVM::canMethodEnterEventBeHooked() || TR::Options::getAOTCmdLineOptions()->getOption(TR_EnableAOTMethodEnter);
```

Comment:
You should not test cmdLineOptions because they are global and thus common for all clients